proto/go_proto_library.bzl

```diff
@@ -131,8 +131,21 @@ def _go_proto_library_gen_impl(ctx):
   dirs = set([s.short_path[:-1-len(s.basename)]
               for s in srcs + protos])
   cmds += ["/bin/mkdir -p %s/%s" % (work_dir, _drop_external(d)) for d in dirs if d]
-  cmds += ["/bin/cp %s %s/%s" % (s.path, work_dir, _drop_external(s.short_path))
-           for s in srcs + protos]
+
+  if ctx.attr.ignore_go_package_option:
+    # Strip the "option go_package" line from the proto file before compiling,
+    # c.f., https://github.com/bazelbuild/rules_go/issues/323
+    #
+    # NOTE: Using sed does not provide a perfect solution, build will break if
+    # the go_package option splits multiple lines. Use with caution.
+    cmds += ["/bin/sed '/^ *option  *go_package/d' %s > %s/%s" % (s.path, work_dir,
+                                                                  _drop_external(s.short_path))
+             for s in srcs]
+    cmds += ["/bin/cp %s %s/%s" % (s.path, work_dir, _drop_external(s.short_path))
```

Comment:
We only apply the stripping to src files, so that common things like "google/protobuf/..." will not be broken.